api/types/swarm/task.go

```diff
@@ -56,10 +56,12 @@ type Task struct {
 
 // TaskSpec represents the spec of a task.
 type TaskSpec struct {
-	// ContainerSpec and PluginSpec are mutually exclusive.
+	// ContainerSpec, AttachmentSpec and PluginSpec are mutually exclusive.
 	// PluginSpec will only be used when the `Runtime` field is set to `plugin`
-	ContainerSpec *ContainerSpec      `json:",omitempty"`
-	PluginSpec    *runtime.PluginSpec `json:",omitempty"`
+	// AttachmentSpec will be used when the `Runtime` field is set to `attachment`
+	ContainerSpec         *ContainerSpec         `json:",omitempty"`
+	PluginSpec            *runtime.PluginSpec    `json:",omitempty"`
+	NetworkAttachmentSpec *NetworkAttachmentSpec `json:",omitempty"`
```

Comment:
I think we may need to error out on swarm create/update if it includes a NetworkAttachmentSpec?

Reply:
Should we have a separate type for inspect vs create? I know we're reusing some types at various locations (sometimes resulting in information being shown that happens to be there, but is not needed)

Reply:
I don't think a separate type is such a good idea. This is the task spec, having the task spec in the inspect output I think is hugely important.